src/ui/DatePicker.js

```diff
@@ -0,0 +1,130 @@
+import _ from 'lodash';
+import moment from 'moment';
+import PropTypes from 'prop-types';
+import React, { Component } from 'react';
+import { Calendar } from 'react-date-range';
+import { FormControl } from 'react-bootstrap';
+import DatePickerOverlay from './DatePickerOverlay';
+import {
+  DEFAULT_DATE_FORMAT,
+} from './DateRangePicker';
+
+class DatePicker extends Component {
+  state = {
+    show: false,
+  };
+
+  constructor(...arrgs) {
+    super(...arrgs);
+
+    this.onToggle = this.onToggle.bind(this);
+    this.onChange = this.onChange.bind(this);
+  }
+
+  shouldComponentUpdate(nextProps, nextState) {
+    const { value, dateFormat, disabled } = this.props;
+    const { show } = this.state;
+
+    return (
+      value !== nextProps.value
+      || dateFormat !== nextProps.dateFormat
+      || disabled !== nextProps.disabled
+      || show !== nextState.show
+    );
+  }
+
+  onToggle() {
+    this.setState(state => ({ ...state, show: !state.show }));
+  }
+
+  onChange(date) {
+    const { onChange } = this.props;
+    onChange(moment(date).startOf('day').toDate());
+  }
+
+  format(value) {
+    const { dateFormat } = this.props;
+    let date;
+
+    if (!value) {
+      date = moment();
+    } else {
+      date = moment(value);
+    }
+
+    return [date.toDate(), date.format(dateFormat)];
+  }
+
+  render() {
+    const {
+      value, disabled, placement, dateFormat,
+    } = this.props;
+    const { show } = this.state;
+
+    const [date, label] = this.format(value);
+
+    return (
+      <span>
+        <FormControl
+          value={label}
+          disabled={disabled}
+          onClick={this.onToggle}
+          onChange={_.identity}
+          ref={(input) => {
+            this.target = input;
+          }}
+        />
+        <DatePickerOverlay
+          container={this}
+          show={show && !disabled}
+          onHide={this.onToggle}
+          placement={placement}
+          // eslint-disable-next-line react/no-find-dom-node
+          target={() => this.target}
+        >
+          <Calendar
+            date={date}
```

Comment:
You've defined minDate and maxDate in the FieldComponent for this and also in the defaultProps, but you're not providing it to the Calendar component. Both are also missing in the propTypes spec.

Reply:
Good catch, I will add a eslint commit hook so we can spot these things early on